lib/domain.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * (C) Copyright 2017 o2r project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const psl = require('psl');
+const Domain = require('../lib/model/domain');
```

Comment:
I don't understand rightaway why you don't just store the domains within the journal object - why is there an extra database "table" for it? Probably a good reason - please explain in the thesis!